### PR TITLE
Federation migration and additional improvements

### PR DESCRIPTION
**Pull request by sampaiodiego**
Closes #15269

### CODE REVIEW

**Comment by sampaiodiego** [comment]
I changed it [to run in steps of 500 users](https://github.com/RocketChat/Rocket.Chat/pull/15336/files#diff-ae2099f5fe2aa31cbdd73b35a89bf84bR21) this way it was able to run without hitting memory limits.

File: server/startup/migrations/v156.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		limit: 500,
+	};
+	const users = Users.find({ federation: { $exists: true } }, options).fetch();
+
+	const ids = users.map((u) => u._id);
+	if (ids.length === 0) {
+		return;
+	}
+
+	Users.update({ _id: { $in: ids } }, { $unset: { federation: 1 } }, { multi: true });
```
Comment (rodrigok) [inline server/startup/migrations/v156.js]: Can't it be just the update?

Comment (sampaiodiego) [reply server/startup/migrations/v156.js]: it was breaking the server.. changed to use `rawCollection` and it worked fine.

File: server/startup/migrations/v156.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+		// Normalize the federation property on all collections
+
+		// Update rooms
+		console.log('Migration: update rooms');
+		Promise.await(Rooms.model.rawCollection().updateMany({
+			federation: { $exists: true },
+		}, {
+			$unset: { federation: 1 },
+		}, { multi: true }));
```
Comment (rodrigok) [inline server/startup/migrations/v156.js]: Is `{multi: true}` necessary?

File: server/startup/migrations/v156.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+		// Update all subscriptions
+		console.log('Migration: update subscriptions');
+		Promise.await(Subscriptions.model.rawCollection().updateMany({
+			federation: { $exists: true },
+		}, {
+			$unset: { federation: 1 },
+		}, { multi: true }));
```
Comment (rodrigok) [inline server/startup/migrations/v156.js]: Is `{multi: true}` necessary?

File: server/startup/migrations/v156.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		}));
+
+		// Update all messages
+		// We will not update the mentions and channels here
+		console.log('Migration: update messages');
+		Promise.await(Messages.model.rawCollection().updateMany({
+			federation: { $exists: true },
+		}, {
+			$unset: { federation: 1 },
+		}, { multi: true }));
```
Comment (rodrigok) [inline server/startup/migrations/v156.js]: Is `{multi: true}` necessary?

File: server/startup/migrations/v156.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+		// Update all uploads
+		console.log('Migration: update uploads');
+		Promise.await(Uploads.model.rawCollection().updateMany({
+			federation: { $exists: true },
+		}, {
+			$unset: { federation: 1 },
+		}, { multi: true }));
```
Comment (rodrigok) [inline server/startup/migrations/v156.js]: Is `{multi: true}` necessary?

**Pull request merged**